### PR TITLE
Fix warning compilation

### PR DESCRIPTION
**Pull request by platinumthinker**
I use new nif erl_drv_output_term instead deprecated driver_output_term

### CODE REVIEW

**Comment by platinumthinker** [comment]
nif  erl_drv_output_term is deprecated since Erlang R16

**Comment by surik** [comment]
We still have Travis build for 15B03. Can we keep compatibility with older versions?

**Comment by platinumthinker** [comment]
I can use ifdef for backward compatibility

**Comment by platinumthinker** [comment]
@surik, ping?

**Comment by platinumthinker** [comment]
@RoadRunnr ping?

File: c_src/gen_socket_drv.c [hunk 1]
```diff
@@ -160,7 +160,11 @@ gs_send_single_atom(GsState* state, ErlDrvTermData atom)
 		ERL_DRV_ATOM, atom,
 		ERL_DRV_TUPLE, 2
 	};
-	driver_output_term(state->drv_port, output, sizeof(output) / sizeof(ErlDrvTermData));
+#ifndef OLD_NIF
+    driver_output_term(state->drv_port, output, sizeof(output) / sizeof(ErlDrvTermData));
```
Comment (RoadRunnr) [inline c_src/gen_socket_drv.c]: indention is wrong

File: c_src/gen_socket_drv.c [hunk 2]
```diff
@@ -160,7 +160,11 @@ gs_send_single_atom(GsState* state, ErlDrvTermData atom)
 		ERL_DRV_ATOM, atom,
 		ERL_DRV_TUPLE, 2
 	};
-	driver_output_term(state->drv_port, output, sizeof(output) / sizeof(ErlDrvTermData));
+#ifndef OLD_NIF
```
Comment (RoadRunnr) [inline c_src/gen_socket_drv.c]: the ifndef logic seem to be wrong, I'm still seeing warning on new OTP versions

**Comment by RoadRunnr** [comment]
sorry, I missed the pull request somehow. Thanks for the reminder.

**Comment by RoadRunnr** [comment]
Could you also try to add ```-Werror``` into the CFLAGS in c_src/Makefile? That should make travis catch the warnings as errors.

**Comment by platinumthinker** [comment]
@RoadRunnr i fixed errors and add compile option -Werror

**Comment by RoadRunnr** [comment]
looks good, please squash the commits into one. Then it should be ready to merge.

**Comment by platinumthinker** [comment]
@RoadRunnr, done

**Comment by platinumthinker** [comment]
@RoadRunnr, ping?

**Pull request merged**